plasmapy/atomic/ionization_state.py

```diff
@@ -136,15 +136,60 @@ def __str__(self) -> str:
         return f"<IonizationState of {symbol}>"
 
     def __repr__(self):
-        return self.__str__()
+        """Show diagnostic information of an IonizationState instance."""
+
+        minimum_ionic_fraction_to_show = 1e-3
+
+        output = f"IonizationState instance of {self.particle}"
+
+        if np.any(np.isnan(self.ionic_fractions)):
+            return output
+
+        Z_mean = "{:.2f}".format(self.Z_mean)
+        output += f'with Z_mean = {Z_mean}\n\n'
+
+        for state in self:
+            if state.ionic_fraction > minimum_ionic_fraction_to_show:
+
+                symbol = state.ionic_symbol
+                if state.integer_charge < 10:
+                    symbol = symbol[:-2] + ' ' + symbol[-2:]
+
+                fraction = "{:.3f}".format(state.ionic_fraction)
+
+                output += (f'  {symbol}: {fraction}')
+
+                if np.isfinite(self.n_elem):
+                    value = "{:.2e}".format(state.number_density.si.value)
+                    output += f"     n_i = {value} m**-3"
+
+                output += '\n'
+
+        if np.isfinite(self.T_e) or np.isfinite(self.n_elem):
+            output += '\n'
+        if np.isfinite(self.T_e):
+            T_e = "{:.2e}".format(self.T_e.value) + " K"
+            output += f'  T_e    = {T_e}\n'
+        if np.isfinite(self.n_elem):
+            n_elem = "{:.2e}".format(self.n_elem.value) + " m**-3"
+            n_e = "{:.2e}".format(self.n_e.value) + " m**-3"
+            output += f'  n_elem = {n_elem}\n'
+            output += f'  n_e    = {n_e}'
+
+        return output
```

Comment:
Oh wow, that's a pretty dense `__repr__`. It's probably no issue here, but it got me thinking... dense enough `__repr__`s (probably in some limit...  that might fall under High Energy Python or something more relative) may actually slow down PyCharm and other "smarter" editors (vim doesn't care) which render `__repr__`s dynamically and often.

Reply:
My motivation here was so that people using this class could quickly diagnose what the ionization state is for this particular element when working on the command line.  
```Python
>>> IonizationState('He', [0.1, 0.9, 0.0], n_e=4e19*u.m**-3, T_e=4e4*u.K)
IonizationState instance of He with Z_mean = 0.90

  He  0+: 0.100     n_i = 4.00e+18 m**-3
  He  1+: 0.900     n_i = 3.60e+19 m**-3

  T_e    = 4.00e+04 K
  n_elem = 4.00e+19 m**-3
  n_e    = 3.60e+19 m**-3
```

Reply:
Oh wait, I just noticed the newlines. That might well crash stuff, though I'm not certain it's actually going to happen. I'll look into it later.

Reply:
Okay, I just checked and this is completely fine. Turns out pycharm and the like use `str`, not `repr`.

![image](https://user-images.githubusercontent.com/11289391/46933553-cffb8e00-d054-11e8-90f2-40d8e69830b2.png)

And here's the result with `str` and `repr` swapped (the gist of it is: nothing blew up):
![image](https://user-images.githubusercontent.com/11289391/46933638-29fc5380-d055-11e8-9cb6-1530a99fca2d.png)